### PR TITLE
DAS-2267: Add CI config for testing and reporting.

### PR DESCRIPTION
**Pull request by flamingbear**
## Description

This PR implements the CI for harmony-SMAP-L2-gridding-service.
It follows HyBIG's model

## Jira Issue ID

[DAS-2267](https://bugs.earthdata.nasa.gov/browse/DAS-2267)

## Local Test Steps

This is probably easiest to test by forking the
[nasa/harmony-SMAP-L2-gridding-service](https://github.com/nasa/harmony-SMAP-L2-gridding-service/tree/main)
repo to your own GitHub repo and running it there to see that all of the checks
run properly and that you can build and publish an image.

That's what I did on my fork: https://github.com/flamingbear/harmony-SMAP-L2-gridding-service/

Here is the results of merging a new PR after these changes were merged to main. (showing the tests, build of the docker image and publishing of the test-results and coverage artifact.)
https://github.com/flamingbear/harmony-SMAP-L2-gridding-service/actions/runs/12057871278

## PR Acceptance Checklist
* [X] Jira ticket acceptance criteria met.
* [X] `CHANGELOG.md` updated to include high level summary of PR changes.
* [X] `docker/service_version.txt` updated if publishing a release.
* [X] Tests added/updated and passing.
* [X] Documentation updated (if needed).

### CODE REVIEW

File: smap_l2_gridder/crs.py
Comment (flamingbear) [inline smap_l2_gridder/crs.py:1]: These changes were slipped in from the previous PR feedback.  It is replacing the long well known text from epsg.org with the epsg code and using the output from pyproj's `to_wkt` there are subtle differences in the output, but the simplicity outweighs those nits.

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    uses: ./.github/workflows/run_service_tests.yml
+
+  run_lib_tests:
+    uses: ./.github/workflows/run_lib_tests.yml
+
+  mypy:
+    uses: ./.github/workflows/mypy.yml
+
+  build_and_publish:
+    needs: [run_service_tests, run_lib_tests, mypy]
```
Comment (owenlittlejohns) [inline .github/workflows/publish_release.yml:32]: Nice! Although I'm sure we'll all be frustrated when `mypy` starts complaining at something insidious in a few months time!

**Review by owenlittlejohns — APPROVED** [review]
Looks good to me!

I'm a big old nerd for `mypy` and am excited to have a project have it baked in from the start.

File: .github/workflows/publish_release.yml (hunk 1 above)
Comment (flamingbear) [reply .github/workflows/publish_release.yml:32]: 💯 I already got bit by it.

**Pull request merged**